master/src/test/java/org/apache/celeborn/service/deploy/master/SlotsAllocatorSuiteJ.java

```diff
@@ -334,4 +334,16 @@ public void testHDFSOnly() {
     final boolean shouldReplicate = true;
     checkSlotsOnHDFS(workers, partitionIds, shouldReplicate, true);
   }
+
+  @Test
+  public void testAllocateSlotsWithNoAvailableSlots() {
+    final List<WorkerInfo> workers = prepareWorkers(true);
+    // Simulates no available slots behavior with greatly changed estimatedPartitionSize for workers
+    // with usable disks.
+    workers.forEach(workerInfo -> workerInfo.updateDiskMaxSlots(Long.MAX_VALUE));
```

Comment:
Set the max slots to Long.MAX_VALUE and offer slots for zero partitions won't test this patch. I think you can test that the max slots are zero and offer slots for some partitions.

Reply:
@FMX, this test cases invoke `updateDiskMaxSlots(estimatedPartitionSize)` to set the max slots to zero and offer slots to partition which id is zero. This test case verifies whether fallback to roundrobin works well with no available slots.